docs/reference/setup-tools/kubeadm/internal-design.md

```diff
@@ -0,0 +1,581 @@
+---
+approvers:
+- mikedanese
+- luxas
+- jbeda
+title: Internal design 
```

Comment:
On a second thought, "Implementation details" or "kubeadm under the hood" or similar could also make sense here